common/djangoapps/microsite_configuration/backends/base.py

```diff
@@ -0,0 +1,101 @@
+"""
+Microsite configuration backend module.
+
+Contains the base class for microsite backends.
+
+"""
+
+from __future__ import absolute_import
+
+import abc
+
+
+# pylint: disable=unused-argument
+class BaseMicrositeBackend(object):
+    """
+    Abstract Base Class for the microsite backends.
+
+    """
+    __metaclass__ = abc.ABCMeta
+
+    def __init__(self, **kwargs):
+        pass
+
+    @abc.abstractmethod
+    def set_config_by_domain(self, domain):
+        """
+        For a given request domain, find a match in our microsite configuration
+        and make it available to the complete django request process
+        """
+        pass
```

Comment:
@felipemontoya Typically I've just thrown a NotImplementedError() in the abstract base classes. Mind if I change?